### PR TITLE
Sort attribution destinations on parsing

### PR DESCRIPTION
**Pull request by apasel422**
***
<a href="https://pr-preview.s3.amazonaws.com/apasel422/attribution-reporting-api/pull/1272.html" title="Last updated on May 7, 2024, 6:24 PM UTC (76b590f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/1272/4d166c8...apasel422:76b590f.html" title="Last updated on May 7, 2024, 6:24 PM UTC (76b590f)">Diff</a>

### CODE REVIEW

File: ts/src/header-validator/event_level_report.test.ts [hunk 1]
```diff
@@ -334,6 +334,36 @@ const testCases: jsontest.TestCase<EventLevelReport>[] = [
...
+      sourceDebugKey: null,
+      sourceType: SourceType.navigation,
+      sourceEventId: 1n,
+      triggerData: 2n,
+      triggerDebugKey: null,
+      triggerSummaryBucket: null,
+    }),
+    expectedWarnings: [
+      {
+        msg: 'although order is semantically irrelevant, list is expected to be sorted',
```
Comment (linnan-github) [inline ts/src/header-validator/event_level_report.test.ts:362]: Is this warning necessary? I don't think we really *expect* the list to be sorted.

Comment (apasel422) [reply ts/src/header-validator/event_level_report.test.ts:362]: Why don't we expect it to be sorted? This is applied to event-level report bodies, not source registrations.

Comment (linnan-github) [reply ts/src/header-validator/event_level_report.test.ts:362]: Ah sorry I misread it as source registrations. Then should this be an error per spec?

Comment (apasel422) [reply ts/src/header-validator/event_level_report.test.ts]: Done.

**Review by linnan-github — APPROVED** [review]

**Pull request merged**